distributed/shuffle/tests/test_shuffle.py

```diff
@@ -654,10 +657,91 @@ def test_processing_chain():
+            ),
+            # Nullable dtypes
+            f"col{next(counter)}": pd.array([True, False] * 50, dtype="boolean"),
+            f"col{next(counter)}": pd.array(range(100), dtype="Int8"),
+            f"col{next(counter)}": pd.array(range(100), dtype="Int16"),
+            f"col{next(counter)}": pd.array(range(100), dtype="Int32"),
+            f"col{next(counter)}": pd.array(range(100), dtype="Int64"),
+            f"col{next(counter)}": pd.array(range(100), dtype="UInt8"),
+            f"col{next(counter)}": pd.array(range(100), dtype="UInt16"),
+            f"col{next(counter)}": pd.array(range(100), dtype="UInt32"),
+            f"col{next(counter)}": pd.array(range(100), dtype="UInt64"),
+            # pandas dtypes
+            f"col{next(counter)}": pd.array(
+                [np.datetime64("2022-01-01") + i for i in range(100)],
+                dtype=pd.DatetimeTZDtype(tz="Europe/Berlin"),
+            ),
+            f"col{next(counter)}": pd.array(
+                [pd.Period("2022-01-01", freq="D") + i for i in range(100)],
+                dtype="period[D]",
+            ),
+            f"col{next(counter)}": pd.array(
+                [pd.Interval(left=i, right=i + 2) for i in range(100)], dtype="Interval"
+            ),
+            f"col{next(counter)}": pd.array(["x", "y"] * 50, dtype="category"),
+            f"col{next(counter)}": pd.array(["lorem ipsum"] * 100, dtype="string"),
+            # PyArrow dtypes
+            f"col{next(counter)}": pd.array([True, False] * 50, dtype="bool[pyarrow]"),
+            f"col{next(counter)}": pd.array(range(100), dtype="int8[pyarrow]"),
+            f"col{next(counter)}": pd.array(range(100), dtype="int16[pyarrow]"),
+            f"col{next(counter)}": pd.array(range(100), dtype="int32[pyarrow]"),
+            f"col{next(counter)}": pd.array(range(100), dtype="int64[pyarrow]"),
+            f"col{next(counter)}": pd.array(range(100), dtype="uint8[pyarrow]"),
+            f"col{next(counter)}": pd.array(range(100), dtype="uint16[pyarrow]"),
+            f"col{next(counter)}": pd.array(range(100), dtype="uint32[pyarrow]"),
+            f"col{next(counter)}": pd.array(range(100), dtype="uint64[pyarrow]"),
+            f"col{next(counter)}": pd.array(range(100), dtype="float32[pyarrow]"),
+            f"col{next(counter)}": pd.array(range(100), dtype="float64[pyarrow]"),
+            f"col{next(counter)}": pd.array(
+                [pd.Timestamp.fromtimestamp(1641034800 + i) for i in range(100)],
+                dtype=pd.ArrowDtype(pa.timestamp("ms")),
+            ),
+            # FIXME: distributed#7420
+            # f"col{next(counter)}": pd.array(
+            #     ["lorem ipsum"] * 100,
+            #     dtype="string[pyarrow]",
+            # ),
+            # f"col{next(counter)}": pd.array(
+            #     ["lorem ipsum"] * 100,
+            #     dtype=pd.StringDtype("pyarrow"),
+            # ),
```

Comment:
My guess is we're running into https://github.com/pandas-dev/pandas/issues/50074 here

Reply:
Do we have access to, or could easily keep track of, the original input dtypes at the point when we convert the `pa.Table` to a `pd.DataFrame`?

Reply:
Left for follow-up work.